### ISSUE TITLE
Models are getting information of the future

### ISSUE THREAD

**Issue by djboblo-99** [Soroush98/LSTM-CNN_Stock#1]
Hi, I came across your repo

in the following section of your code you are providing information about the future to the model:

`for i in range(0 , len(stock) - window_size -1 , 1):
    first = stock.iloc[i, 4]
    temp = []
    temp2 = []
    for j in range(window_size):
        temp.append((stock.iloc[i + j, 4] - first) / first)
    # for j in range(week):
    temp2.append((stock.iloc[i +window_size, 4] - first) / first)
    # X.append(np.array(stock.iloc[i:i+window_size,4]).reshape(50,1))
    # Y.append(np.array(stock.iloc[i+window_size,4]).reshape(1,1))
    # print(stock2.iloc[i:i+window_size,4])
    X.append(np.array(temp).reshape(50, 1))
    Y.append(np.array(temp2).reshape(1,1)))`

for i = 0 temp gets the first 50 close values of the csv
these are beeing saved in the X variable
but you also save the 50th close value in temp2 and therefore in Y
This means that the model already has the value we want to predict in Y

Also the accuracy the model has after training for only 1 epoch is to good to be true

**Comment by Soroush98** [comment]
Hi , Thanks for your comment, in first iteration close price from 0 to (window_size-1) will be saved in X and Y will have the (windows_size) close price. so always the future close price we want to predict will be in Y, then with these information we will train our model to predict the next day of a current window given

**Comment by djboblo-99** [comment]
Ok, my bad I just checked it and you are right.
But it still seems to me that the model isn't predicting and doing more of a comparison.
Your dense Layer is size 1 and stateful = false
So prediction is only 1 step ahead?

What do you think of this answer:
https://stackoverflow.com/questions/48760472/how-to-use-a-keras-rnn-model-to-forecast-for-future-dates-or-events


I don't mean to criticise, just trying to understand your code :)

**Comment by Soroush98** [comment]
I appreciate it , Yeah you are right I forgot to commit new changes to my code which has stateful = true, When you are using batch size for time series , stateful should be true. Dense 1 on the last layer is because we want to predict only a single step ahead.

**Comment by djboblo-99** [comment]
Can you maybe commit the new code?
Thanks!
